### ISSUE TITLE
Apostrophes in doc_markdown

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#2210]
```rust
pub struct Collection { /* ... */ }

/// An iterator over mycrate::Collection's values.
pub struct Iter { /* ... */ }
```

```
warning: you should put `mycrate::Collection's` between ticks in the documentation
   |
   | /// An iterator over mycrate::Collection's values.
   |                      ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(doc_markdown)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.169/index.html#doc_markdown
```

I would expect this to recommend ``` `mycrate::Collection`'s ``` instead of ``` `mycrate::Collection's` ```.

**Comment by mockersf** [comment]
issue is [here](https://github.com/rust-lang-nursery/rust-clippy/blob/master/clippy_lints/src/doc.rs#L239)
```
let word = word.trim_matches(|c: char| !c.is_alphanumeric());
```
